src/span_context.js

```diff
@@ -68,7 +68,11 @@ export default class SpanContext {
 
   get traceId(): any {
     if (this._traceId == null && this._traceIdStr != null) {
-      this._traceId = Utils.encodeInt64(this._traceIdStr);
+      const safeTraceIdStr = this._traceIdStr.length % 2 == 0 ? this._traceIdStr : '0' + this._traceIdStr;
+      const tmpBuffer = Utils.newBuffer(safeTraceIdStr, 'hex');
+      const size = tmpBuffer.length > 8 ? 16 : 8;
+      this._traceId = Utils.newBuffer(size);
```

Comment:
why do we need to allocate two buffers?

Reply:
I need to create a 64 or 128 bits buffer and decode the hex string at the same, I don't think that can be done in one shot

Reply:
It was unfortunate oversight that Jaeger clients allowed shortened string representation of trace/span IDs by dropping leading zeroes (an optimization that only affects ~1/16th of all traces, wasn't worth it), instead of always insisting on exactly 16 or 32 characters. I booked a meta-issue for this: https://github.com/jaegertracing/jaeger/issues/1657

To maintain backwards compatibility we do need to be able to accept strings <16 or <32 chars, but always output strings of the exact length.

The overall implementation would probably be simpler if internal representation always uses a buffer of exact length, as you're doing above (an alternative would be use use the buffer of whatever length the input string is). But we should still avoid double-allocating the buffer. For example, in L73 instead of padding  with a single zero, you could pad with as many zeros as needed to reach 16/32 length, and then your `Utils.newBufferFromHex()` function will return the right-sized buffer in one allocation. We still lose an allocation due to padding, but that will only affect about 1/16th of all traces, and once https://github.com/jaegertracing/jaeger/issues/1657 is fixed for all languages padding won't be happening in practice.